packages/sitecore-jss/src/site/graphql-siteinfo-service.ts

```diff
@@ -128,6 +160,15 @@ export class GraphQLSiteInfoService {
       return [];
     }
 
+    const results: SiteInfo[] = this.config.xmCloud
```

Comment:
Currently, if you try to scaffold nextjs-xmcloud + nextjs-multisite
the GraphQLSiteInfoService will use SXP query still, so it will be required for customer to manually set this property that is not the best dev experience.
Instead I can suggest you to set: "xmCloud: true" conditionally using ejs template, [here](https://github.com/Sitecore/jss/blob/dev/packages/create-sitecore-jss/src/templates/nextjs-multisite/scripts/config/plugins/multisite.ts#L20)
If "nextjs-xmcloud" base template is applied, you will set it

Reply:
Additionally, I'd recommend the config option not be platform-specific, but descriptive of the method used to fetch site info. Maybe `useSiteQuery`. This would technically still make sense then when XP 10.4 is released.

Reply:
> Additionally, I'd recommend the config option not be platform-specific

I don't think that's the best approach. We enable site query as an option temporary - before we switch to it completely some time in April. We need to be explicit about this functionality only working on XMCloud currently.

Reply:
@art-alexeyenko Our goal should be to keep platform awareness out of the base packages (which we have thus far), and only concern themselves with the API they are wrapping / consuming (in this case GraphQL Edge schema). For example, elsewhere in our packages we have functionality to deal with embedded personalization. However, we don't explicitly have any "xmcloud" flags. That is reserved for the initializer/app layer.

I do agree there does need to be a heads up raised for this option, but I think `site` query compatibility can easily be explained in a JSDoc comment on the `useSiteQuery` option (e.g. "`site` query is available on XM Cloud and XP 10.4+").

Reply:
@ambrauer done